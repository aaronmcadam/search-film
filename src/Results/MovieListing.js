import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Movie from '../Movie';

const Wrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
`;

function MovieListing({ movies }) {
  if (!movies.length) {
    return <p>We couldn't find anything</p>;
  }

  return (
    <Wrapper>
      {movies.map(movie => <Movie key={movie.id} {...movie} />)}
    </Wrapper>
  );
}

MovieListing.propTypes = {
  movies: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      poster_path: PropTypes.string,
      title: PropTypes.string,
      overview: PropTypes.string,
    })
  ),
};

MovieListing.defaultProps = {
  movies: [],
};

export default MovieListing;
