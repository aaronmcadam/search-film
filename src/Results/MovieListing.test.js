import React from 'react';
import { shallow } from 'enzyme';
import MovieListing from './MovieListing';

const setup = propOverrides => {
  const props = {
    movies: [],
    ...propOverrides,
  };
  const wrapper = shallow(<MovieListing {...props} />);

  return {
    wrapper,
  };
};

test('renders', () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});
