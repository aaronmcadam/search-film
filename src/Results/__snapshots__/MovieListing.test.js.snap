// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<p>
  We couldn't find anything
</p>
`;
